File: csharp/4.cs
using System;

public class Solution {

  public static bool IsPalindrome(ListNode head) {
    StringBuilder sb = new StringBuilder();
    ListNode current = head;

    while (current != null) {
      sb.Append(current.val);
      current = current.next;
    }
       
    string items = sb.ToString();
    string reversed = new string(items.ToCharArray().Reverse().ToArray());
    return reversed == items;
  }
}

File: javascript/4.js
class ListNode {
   constructor(val, next = null) {
       this.val = val;
       this.next = next;
   }
}

function isPalindrome(head) {
   let vals = [];
   let current = head;
   while (current !== null) {
       vals.push(current.val);
       current = current.next;
   }
   return vals.join('') === vals.slice().reverse().join('');
}

let head = new ListNode(1, new ListNode(2, new ListNode(2, new ListNode(1))));
console.log(isPalindrome(head));
 